matlab/valid_setpts.m

```diff
@@ -1,37 +1,45 @@
-function [nj, nk] = valid_setpts(type,dim,x,y,z,s,t,u)
+function [nj, nk] = valid_setpts(is_gpuarray,type,dim,x,y,z,s,t,u)
 % VALID_SETPTS   validate guru finufft_setpts input sizes for a type and dim
+%                is_gpuarray should be 1 if data is expected on gpu, or 0 if on cpu
 %
-% [nj nk] = valid_setpts(type,dim,x,y,z,s,t,u) raises errors if there are
+% [nj nk] = valid_setpts(is_gpuarray,type,dim,x,y,z,s,t,u) raises errors if there are
 %  incompatible input sizes for transform type (1, 2 or 3) and dimension dim,
-%  and returns nj (aka M), and, for type 3, also nk (aka N). The returned
+%  and returns nj (aka M), and, for type 3, also nk (aka N). It also checks if the arrays
+%  are on the expected device, is_gpuarray 0 for cpu, 1 for gpu. The returned
 %  values are int64 (not the usual double class of numel).
 %
-% nj = valid_setpts(type,dim,x,y,z) is also allowed for types 1, 2.
+% nj = valid_setpts(is_gpuarray,type,dim,x,y,z) is also allowed for types 1, 2.
 
 % Barnett 6/19/20, split out from guru so simple ints can check before plan.
 % s,t,u are only checked for type 3.
-% note that isvector([]) is false.
+% note that isvector([]) is false, finufft_isgpuarray([]) is false, finufft_isgpuarray(gpuArray([])) if true.
```

Comment:
I have a typo here, I think I mean `is true` not `if true`. I will correct this.